From e85736def5ce0c89a7dd1c05d9071a31feed05cc Mon Sep 17 00:00:00 2001
From: Richard <xplora@wak.co.nz>
Date: Thu, 1 May 2025 14:12:22 +1200
Subject: [PATCH] Update cyberduck.sh

Horribly bad form to download the same file twice, because 1. it's very unfriendly to NAT'd environments as the server might see the extra hit as cause for alarm, and 2. there is the tenet if chances the information could change between the two downloads, resulting in the downloadURL containing a link to an older version of the app than the version that then gets identified in appNewVersion.
---
 fragments/labels/cyberduck.sh | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/fragments/labels/cyberduck.sh b/fragments/labels/cyberduck.sh
index 3f57fe52f..46d3c068c 100644
--- a/fragments/labels/cyberduck.sh
+++ b/fragments/labels/cyberduck.sh
@@ -1,7 +1,8 @@
 cyberduck)
     name="Cyberduck"
     type="zip"
-    downloadURL=$(curl -fs https://version.cyberduck.io/changelog.rss | xpath '//rss/channel/item/enclosure/@url' 2>/dev/null | cut -d '"' -f 2 )
-    appNewVersion=$(curl -fs https://version.cyberduck.io/changelog.rss | xpath '//rss/channel/item/enclosure/@sparkle:shortVersionString' 2>/dev/null | cut -d '"' -f 2 )
+    changeLOG="$( curl -fs https://version.cyberduck.io/changelog.rss )"
+    downloadURL=$( echo "$changeLOG" | xpath '//rss/channel/item/enclosure/@url' 2>/dev/null | cut -d '"' -f 2 )
+    appNewVersion=$( echo "$changeLOG" | xpath '//rss/channel/item/enclosure/@sparkle:shortVersionString' 2>/dev/null | cut -d '"' -f 2 )
     expectedTeamID="G69SCX94XU"
     ;;
